src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -423,5 +424,83 @@ private static string NormalizeDateTimeCollectionItems(string bracketLiteralText
 
             return "[" + String.Join(",", items) + "]";
         }
+
+        private static bool IsCollectionContentEmptyOrSpaces(string bracketLiteralText)
+        {
+            if (string.IsNullOrWhiteSpace(bracketLiteralText) || bracketLiteralText.Length < 2)
+            {
+                return true;
+            }
+
+            string content = bracketLiteralText[1..^1].Trim();
+
+            if (string.IsNullOrWhiteSpace(content))

```

Comment:
Depending on the answer to this https://github.com/OData/odata.net/pull/3190/files#r1978721142, is this scenario possible?

I think that the first thing we should check is `string.IsNullOrWhiteSpace` at the beginning of this method

Reply:
There is a check for that at the beginning of the method. This is after removing the first and last brackets and trimmed spaces before and after content.